src/viur_cli/conf.py

```diff
@@ -1,202 +1,174 @@
         """
-    global projectConfig
 
-    if not path:
-        path = projectConfigFilePath
 
         # Search in any parent folder for a project.json,
         # change working directory because subsequent commands
         # require for project root folder.
         changed = False
-        while not os.path.exists(path):
+        while not os.path.exists(PROJECT_CONFIG_FILE):
             os.chdir("..")
             changed = True
 
             if os.getcwd() == "/":
-                echo_fatal(f"{path} not found - please check if you are in the right folder.")
+                echo_fatal(f"{PROJECT_CONFIG_FILE} not found - please check if you are in the right folder.")
 
         if changed:
             echo_info(f"Project root is {os.getcwd()}")
 
-    try:
-        f = open(path, "r")
-        projectConfig = json.loads(f.read())
-    except FileNotFoundError:
-        echo_fatal(f"Can't open {path} for reading")
-
-    except json.decoder.JSONDecodeError as e:
-        echo_fatal(f"The configuration in {path} contains invalid JSON: {str(e)}. Please verify right syntax.")
-
-    update_config(path)
-    return projectConfig
-
-
-def write_config(conf, path=None):
-    """
-    Write the current projectConfig dictionary to project.json.
+        try:
+            f = open(PROJECT_CONFIG_FILE, "r")
+            self.update(json.loads(f.read()))
 
-    This function is used to write the current project configuration, provided as a dictionary,
-    to the project.json file.
+        except FileNotFoundError:
+            echo_fatal(f"Can't open {PROJECT_CONFIG_FILE} for reading")
 
-    :param conf: dict
-        The project configuration to be written to the project.json file.
+        except json.decoder.JSONDecodeError as e:
+            echo_fatal(f"The configuration in {PROJECT_CONFIG_FILE} contains invalid JSON: {str(e)}. Please verify right syntax.")
 
-    :param path: str, optional
-        The path to the project.json file. If not provided, the default projectConfigFilePath is used.
-
-    :return: None
-    """
-    global projectConfig
-
-    projectConfig = conf
-
-    if not path:
-        path = projectConfigFilePath
-
-    f = open(path, "w")
-    f.write(json.dumps(conf, indent=4, sort_keys=True))
-    f.close()
+        self.migrate()
 
+    def save(self):
+        """
+        Write the current projectConfig dictionary to project.json.
+        """
+        f = open(PROJECT_CONFIG_FILE, "w")
+        f.write(json.dumps(self, indent=4, sort_keys=True))
+        f.close()
 
-def get_config():
-    """
-        Get the current project configuration.
+    def get_profile(self, profile):
+        if profile == "format":
+            echo_fatal("Your profile can not be named 'Format' ")
+        return self["default"].copy() | self[profile]
 
-        This function returns the current project configuration stored in the global projectConfig variable.
+    def yeet(self):
```

Comment:
yeet?

Reply:
sorry i made a genZ joke here. it a delete function for entries and profiles, i forgot to rename it.